Source/WebKit/UIProcess/PageLoadState.cpp

```diff
@@ -42,6 +42,11 @@ PageLoadState::PageLoadState(WebPageProxy& webPageProxy)
 
 PageLoadState::~PageLoadState()
 {
+#if ASSERT_ENABLED
+    m_observers.removeAllMatching([](auto& observer) {
+        return !observer;
+    });
```

Comment:
We could also remove `nullptr` observers in `PageLoadState::callObserverCallback()`  Not sure if that will catch them all, though, to make the assertion below never fire.